tests/test_delb_plugin.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+
+
+from delb import Document
+from snakesist import ExistClient
+
+
+@pytest.mark.usefixtures("db")
+def test_delete_document():
+    url = "existdb://admin:@localhost:8080/test_collection/test_document.xml"
+    document = Document(url)
+    document.existdb_delete()
+
+
+@pytest.mark.usefixtures("db")
+def test_load_document_from_url():
+    url = "existdb://admin:@localhost:8080/test_collection/test_document.xml"
+    document = Document(url)
+
+    assert document.config.source_url == url
```

Comment:
in #15 Frank wrote:

> should this property just be bound to the `Document` instances, e.g. `document.source_url`? this here was modeled after the `https_loader`'s behaviour. the interfaces don't allow loaders to interact with the document instance though.

Theo then replied:

> I hope I'm not missing something here, but I don't see why `source_url` couldn't be bound to a `Document` instance. I'm not sure I understand the way the loaders work, I'll have to take a closer look at that.

Reply:
first of all it's a hen-egg-problem. because the intended behaviour is to instantiate a `Document` instead of calling a loader that returns a `Document`. i want to stick with it for aesthethical and practical reasons. the latter manifests in the possibility to pass arbitrary other data to `Document` as keywords argument, like `existdb_client`. so, `Document.__init__` has to call the loader when it isn't completely initialized yet. an easy solution would be to extend loaders' capabilities to also return additional members for the instance.

in case of a plugin that provides both, loader and extension class, the latter could provide such property on the `Document` class as proxy to the `config`'s member. but that doesn't seem sensible with such property that is rather general.

currently i'm thinking whether some general properties may be defined that loaders *can* populate. either by putting them in the `config` object or said additional return value. makes that sense? which mechanic seems simpler to developers?

Reply:
addendum: yes, the not fully initialized document instance could also be passed to the loader. but i rather prefer defined interfaces over a loader possibly messing up the `Document` namespace. yeah, so unpythonic that sentiment.

Reply:
I'm afraid I'm at a loss with this problem right now. I should maybe try to picture some concrete usage examples to make more sense of it.

Reply:
i addressed this in another approach in https://github.com/funkyfuture/delb/issues/10